text/0000-default-field-values.md

```diff
@@ -0,0 +1,1767 @@
+- Feature Name: `default_field_values`
+- Start Date: 2024-08-22
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Allow `struct` definitions to provide default values for individual fields and
+thereby allowing those to be omitted from initializers. When deriving `Default`,
+the provided values will then be used. For example:
+
+```rust
+#[derive(Default)]
+struct Pet {
+    name: Option<String>, // impl Default for Pet will use Default::default() for name
+    age: i128 = 42, // impl Default for Pet will use the literal 42 for age
+}
+```
+
+These can then be used in the following way with the Functional Record Update syntax, with no value:
```

Comment:
I think it's useful if the RFC draws parallel to the FRU syntax whenever possible, to avoid needing to re-explain things like how struct initializer expressions are type checked (especially bc going into the detail of how it actually works means this RFC is gonna get it wrong). But it's worthwhile calling this something different than FRU.